Lib/fontTools/feaLib/ast.py

```diff
@@ -229,14 +229,21 @@ def asFea(self, indent=""):
         return res
 
 
-class FeatureNamesBlock(Block):
-    def __init__(self, location):
+class NestedBlock(Block):
+    def __init__(self, location, tag, block_name):
         Block.__init__(self, location)
+        self.tag = tag
+        self.block_name = block_name
+
+    def build(self, builder):
+        Block.build(self, builder)
+        if self.block_name == "ParamUILabelNameID":
+            builder.add_to_cv_num_named_params(self.tag)
 
     def asFea(self, indent=""):
-        res = indent + "featureNames {\n"
+        res = "{}{} {{\n".format(indent, self.block_name)
```

Comment:
this is an instance where the new-style string formatting looks less readable than the old `%` style. I guess it comes down to taste. I never got into the `.format()` mini-language.. Anyway, i'm digressing. Whatever works.

Reply:
yes, I agree. In this instance it was more to make it consistent with code in `asFea` methods of other classes.

Reply:
it's fine